wallet/src/libwallet/internal/selection.rs

```diff
@@ -253,7 +262,7 @@ where
 		amount,
 		current_height,
 		minimum_confirmations,
-		max_outputs,
+		calculate_max_inputs_in_block(change_outputs),
```

Comment:
Apologies if this is a misunderstanding, I'm not too familiar with this code. But shouldn't you calculate the max number of inputs for `change_outputs + 1` to account for the receiver's output?

Reply:
This is correct, but first we need understand to we need change. First, we calculate assuming that there will be no change (this gives us also the lowest fee). We check do we have `fee+amount` on our balance or not and only then we figure out do we need a change output or we can transfer without it.    
If we need one - we set:
https://github.com/mimblewimble/grin/blob/6a6a423f3ff8339ac1d52acec1f08d49622e2be9/wallet/src/libwallet/internal/selection.rs#L301
and then select coins using `num_outputs`
https://github.com/mimblewimble/grin/blob/6a6a423f3ff8339ac1d52acec1f08d49622e2be9/wallet/src/libwallet/internal/selection.rs#L322-L330 

At least that's how I get it, probably @yeastplume can verify does this make sense or not.

Reply:
Yes, correct, that's what happens.